csharp/src/Microsoft.ML.OnnxRuntime/OrtThreadingOptions.Shared.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// This class allows to specify global thread pool options
+    /// when instantiating the ONNX Runtime environment for the first time.
+    /// </summary>
+    public class OrtThreadingOptions : SafeHandle
+    {
+        /// <summary>
+        /// A pointer to a underlying native instance of ThreadingOptions
+        /// </summary>
+        internal IntPtr Handle => handle;
+        
+        /// <summary>
+        /// Create an empty threading options.
+        /// </summary>
+        public OrtThreadingOptions() : base(IntPtr.Zero, true)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateThreadingOptions(out handle));
+        }
+
+        /// <summary>
+        /// Set global inter-op thread count.
+        /// Setting it to 0 will allow ORT to choose the number of threads used for parallelization of
+        /// multiple kernels. Setting it to 1 will cause the main thread to be used (i.e., no thread pools will be used).
```

Comment:
nit: should call out that the execution mode must be parallel for this value to do/mean anything